Jellyfin.Api/Controllers/DynamicHlsController.cs

```diff
@@ -1432,7 +1432,7 @@ private async Task<ActionResult> GetDynamicSegment(StreamingRequestDto streaming
             var cancellationTokenSource = new CancellationTokenSource();
             var cancellationToken = cancellationTokenSource.Token;
 
-            using var state = await StreamingHelpers.GetStreamingState(
+            var state = await StreamingHelpers.GetStreamingState(
```

Comment:
When should they be disposed?

Reply:
Once the transcoding job is killed or finished.

Reply:
Wouldn't something like this be better then?

```diff
Index: Jellyfin.Api/Models/StreamingDtos/StreamState.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Jellyfin.Api/Models/StreamingDtos/StreamState.cs b/Jellyfin.Api/Models/StreamingDtos/StreamState.cs
--- a/Jellyfin.Api/Models/StreamingDtos/StreamState.cs	(revision bece161ff09a80b6aa86ac2872056422faebd34c)
+++ b/Jellyfin.Api/Models/StreamingDtos/StreamState.cs	(date 1656052011415)
@@ -47,11 +47,6 @@
             }
         }
 
-        /// <summary>
-        /// Gets or sets the transcoding throttler.
-        /// </summary>
-        public TranscodingThrottler? TranscodingThrottler { get; set; }
-
         /// <summary>
         /// Gets the video request.
         /// </summary>
@@ -191,11 +186,8 @@
                 {
                     _mediaSourceManager.CloseLiveStream(MediaSource.LiveStreamId).GetAwaiter().GetResult();
                 }
-
-                TranscodingThrottler?.Dispose();
             }
 
-            TranscodingThrottler = null;
             TranscodingJob = null;
 
             _disposed = true;
Index: Jellyfin.Api/Helpers/TranscodingJobHelper.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Jellyfin.Api/Helpers/TranscodingJobHelper.cs b/Jellyfin.Api/Helpers/TranscodingJobHelper.cs
--- a/Jellyfin.Api/Helpers/TranscodingJobHelper.cs	(revision bece161ff09a80b6aa86ac2872056422faebd34c)
+++ b/Jellyfin.Api/Helpers/TranscodingJobHelper.cs	(date 1656052003966)
@@ -654,8 +654,7 @@
         {
             if (EnableThrottling(state))
             {
-                transcodingJob.TranscodingThrottler = state.TranscodingThrottler = new TranscodingThrottler(transcodingJob, new Logger<TranscodingThrottler>(new LoggerFactory()), _serverConfigurationManager, _fileSystem);
-                state.TranscodingThrottler.Start();
+                transcodingJob.TranscodingThrottler = new TranscodingThrottler(transcodingJob, new Logger<TranscodingThrottler>(new LoggerFactory()), _serverConfigurationManager, _fileSystem);
             }
         }

Reply:
Nope. This doesn't work. I've tried this before.

`TranscodingPosition` required by Throttler will stuck at the initial position reported by `OnTranscodeBeginning`.

https://github.com/jellyfin/jellyfin/blob/130c935bc39a584471c828ed6e1a71dae999e6fd/Jellyfin.Api/Helpers/TranscodingJobHelper.cs#L565

Reply:
Why will it get stuck? StreamState.TranscodeThrottler is not read by anything

Reply:
But my proposed change only removes the throttler from StreamState so it doesn't get disposed

Reply:
Done.